rayon-core/src/configuration.rs

```diff
@@ -1,53 +1,36 @@
 #[allow(unused_imports)]
 use log::Event::*;
 use std::any::Any;
+use std::env;
 use std::error::Error;
+use std::str::FromStr;
 use std::fmt;
 use std::sync::Arc;
 use registry;
+use num_cpus;
 
-/// Custom error type for the rayon thread pool configuration.
+/// Error if the gloal thread pool is initialized multiple times.
 #[derive(Debug,PartialEq)]
-pub enum InitError {
-    /// Error if number of threads is set to zero.
-    NumberOfThreadsZero,
+pub struct GlobalPoolAlreadyInitialized;
```

Comment:
maybe we should move this into the `registry.rs` module and make it private there (and just have the register already convert to `Box<Error>`). Having it public here makes me nervous, and it doesn't seem to be locally used anywhere anyhow.